### PR TITLE
Fix genindex.html to satisfy xhtml standard

### PR DESCRIPTION
**Pull request by shibukawa**
Reopen PR #2836 to master

### CODE REVIEW

**Comment by tk0miya** [comment]
I think you need to remove unused CSS.

```
$ grep -r indextable sphinx | grep -e dl -e dt -e dd | grep css
sphinx/themes/agogo/static/agogo.css_t:table.indextable dl, table.indextable dd {
sphinx/themes/basic/static/basic.css_t:table.indextable dl, table.indextable dd {
sphinx/themes/epub/static/epub.css:table.indextable dl, table.indextable dd {
sphinx/themes/nonav/static/nonav.css:table.indextable dl, table.indextable dd {
sphinx/themes/scrolls/static/scrolls.css_t:table.indextable dl dd {
sphinx/themes/scrolls/static/scrolls.css_t:table.indextable dl dd a {
sphinx/themes/traditional/static/traditional.css_t:table.indextable dl, table.indextable dd {
```

**Comment by tk0miya** [comment]
It seems margins for lists are appeared. They are removed by the styles of themes.
Did you intend this?

## before (alabaster)



## after (alabaster)



## before (classic)



## after (classic)

File: sphinx/themes/basic/genindex-single.html [hunk 1]
```diff
@@ -36,18 +34,18 @@ <h1 id="index">{% trans key=key %}Index &ndash; {{ key }}{% endtrans %}</h1>
...
 <table style="width: 100%" class="indextable"><tr>
   {%- for column in entries|slice(2) if column %}
-  <td style="width: 33%" valign="top"><dl>
+  <td style="width: 33%; vertical-align: top;"><ul style="list-style-type: none;">
     {%- for entryname, (links, subitems, _) in column %}
-      {{ indexentries(entryname, links) }}
+      <li>{{ indexentries(entryname, links) }}</li>
       {%- if subitems %}
-      <dd><dl>
+      <ul style="list-style-type: none;">
```
Comment (tk0miya) [inline sphinx/themes/basic/genindex-single.html]: Why don't you use any CSS-class here?

File: CHANGES [hunk 2]
```diff
@@ -129,6 +129,7 @@ Bugs fixed
 * #2775: Fix failing linkcheck with servers not supporting identidy encoding
 * #2833: Fix formatting instance annotations in ext.autodoc.
 * #1911: ``-D`` option of ``sphinx-build`` does not override the ``extensions`` variable
+* Fix ``genindex.html`` to satisfy xhtml standard.
```
Comment (tk0miya) [inline CHANGES]: This should be moved to the v1.5 section.

**Comment by shibukawa** [comment]
Fix style and CSSs

**Comment by tk0miya** [comment]
Thank you always!

**Pull request merged**